### PR TITLE
gdk-pixbuf: relocatable shared lib on macOS + modernize more for conan v2

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

File: recipes/gdk-pixbuf/all/conanfile.py [hunk 1]
```diff
@@ -64,12 +60,23 @@ def configure(self):
     def layout(self):
         basic_layout(self, src_folder="src")
 
+    def requirements(self):
+        self.requires("glib/2.75.2")
```
Comment (SpaceIm) [inline recipes/gdk-pixbuf/all/conanfile.py:64]:
```suggestion
        self.requires("glib/2.75.3")
```

**Review by jwillikers — DISMISSED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — DISMISSED** [review]

**Review by jwillikers — APPROVED** [review]

**Comment by SpaceIm** [comment]
```
ERROR: Package 'zstd/1.5.4' not resolved: Error downloading file https://c3i.jfrog.io/c3i/api/conan/conan-center/v2/conans/zstd/1.5.4/_/_/revisions/15ac0bc440f89247276d78ea4fdf8b14/files/conan_export.tgz: 'HTTPSConnectionPool(host='c3i.jfrog.io', port=443): Max retries exceeded with url: /c3i/api/conan/conan-center/v2/conans/zstd/1.5.4/_/_/revisions/15ac0bc440f89247276d78ea4fdf8b14/files/conan_export.tgz (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0000022DE6C16A08>: Failed to establish a new connection: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond'))'. [Remote: conan-center]
```

@danimtb @uilianries what's wrong here with zstd?

File: recipes/gdk-pixbuf/all/conanfile.py [hunk 2]
```diff
@@ -59,17 +56,31 @@ def configure(self):
...
         if self.options.shared:
-            self.options["glib"].shared = True
+            wildcard = "" if Version(conan_version) < "2.0.0" else "/*"
+            self.options[f"glib{wildcard}"].shared = True
 
     def layout(self):
         basic_layout(self, src_folder="src")
 
+    def requirements(self):
+        self.requires("glib/2.76.0", transitive_headers=True, transitive_libs=True, run=can_run(self))
```
Comment (prince-chrismc) [inline recipes/gdk-pixbuf/all/conanfile.py:66]: The point isnt to let people use it


```suggestion
        self.requires("glib/2.76.0", transitive_headers=True, run=can_run(self))
```

Comment (SpaceIm) [reply recipes/gdk-pixbuf/all/conanfile.py:66]: `glib` is a public dependency of `gdf-pixbuf`, it doesn't make any sense to me to remove `transitive_libs=True`.

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**